bundle/tests/environment_git_test.go

```diff
@@ -1,22 +1,27 @@
 package config_tests
 
 import (
+	"context"
 	"fmt"
 	"strings"
 	"testing"
 
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config/mutator"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestGitAutoLoadWithEnvironment(t *testing.T) {
```

Comment:
We no longer need tests for environment overrides if we move it out of the load phase. The environments are resolves by the time the code gets to the initialize phase.

Reply:
Can you explain why we needed this before and don't need this anymore? This PR does not change overall behavior, just reorders some operations.

In any case, if we think this test is not useful anymore I propose to clean it up in a follow up PR. In this PR it serves as proof that whatever it tests was not broken by this change.

Reply:
> Can you explain why we needed this before and don't need this anymore? 

Environments / Targets are not resolved in the `DefaultMutators()`. They are resolved right afterwards in the loading process. Since before we loaded the Git details in the loading phase it was relevant to test the environment. 

The `environments`/`targets` block is already nil by the time the code gets to initialize phase.